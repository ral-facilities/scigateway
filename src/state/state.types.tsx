import { ThunkAction } from 'redux-thunk';
import { AnyAction } from 'redux';
import { Step } from 'react-joyride';
import {
  ApplicationStrings,
  PluginConfig,
  FeatureSwitches,
  ScheduledMaintenanceState,
  MaintenanceState,
} from './scigateway.types';
import { RouterState } from 'connected-react-router';

export interface Plugin {
  name: string;
  src: string;
  enable: boolean;
  unauthorised?: boolean;
  location: 'main' | 'left' | 'right';
}

export interface ScigatewayNotification {
  message: string;
  severity: string;
}

export interface ScigatewayState {
  notifications: ScigatewayNotification[];
  plugins: PluginConfig[];
  drawerOpen: boolean;
  siteLoading: boolean;
  showHelp: boolean;
  helpSteps: Step[];
  authorisation: AuthState;
  res?: ApplicationStrings;
  features: FeatureSwitches;
  analytics?: AnalyticsState;
  darkMode: boolean;
  highContrastMode: boolean;
  homepageUrl?: string;
  logo?: string;
  scheduledMaintenance: ScheduledMaintenanceState;
  maintenance: MaintenanceState;
  navigationDrawerLogo?: LogoState;
  adminPageDefaultTab?: 'maintenance' | 'download';
  contactUsAccessibilityFormUrl?: string;
  primaryColour?: string;
}

export interface StateType {
  scigateway: ScigatewayState;
  router: RouterState;
}

export interface ActionType<T> {
  type: string;
  payload: T;
}

export type ThunkResult<R> = ThunkAction<R, StateType, null, AnyAction>;

export interface User {
  username: string;
  isAdmin: boolean;
  avatarUrl: string;
}

export interface ICATAuthenticator {
  mnemonic: string;
  keys: { name: string; hide?: boolean }[];
  friendly?: string;
  admin?: boolean;
}

export interface AuthProvider {
  isLoggedIn: () => boolean;
  isAdmin: () => boolean;
  logOut: () => void;
  logIn: (username: string, password: string) => Promise<void>;
  verifyLogIn: () => Promise<void>;
  refresh: () => Promise<void>;
  fetchScheduledMaintenanceState?: () => Promise<ScheduledMaintenanceState>;
  setScheduledMaintenanceState?: (
    scheduledMaintenanceState: ScheduledMaintenanceState
  ) => Promise<string | void>;
  fetchMaintenanceState?: () => Promise<MaintenanceState>;
  setMaintenanceState?: (
    maintenanceState: MaintenanceState
  ) => Promise<string | void>;
  redirectUrl: string | null;
  authUrl: string | undefined;
  user: User | null;
  mnemonic?: string;
  autoLogin?: () => Promise<void>;
}

export interface AuthState {
  failedToLogin: boolean;
  signedOutDueToTokenInvalidation: boolean;
  loading: boolean;
  provider: AuthProvider;
}

export interface AnalyticsState {
  id: string;
  initialised: boolean;
}

export interface LogoState {
  light: string;
  dark: string;
  altTxt: string;
}
