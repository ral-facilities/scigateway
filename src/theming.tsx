import {
  createTheme,
  ThemeProvider,
  StyledEngineProvider,
  Theme,
} from '@mui/material/styles';
import 'react-redux-toastr/lib/css/react-redux-toastr.min.css';
import React from 'react';
import { StateType } from './state/state.types';
import { connect, useSelector } from 'react-redux';
import { checkboxClasses } from '@mui/material/Checkbox';
import {
  formHelperTextClasses,
  formLabelClasses,
  inputClasses,
  outlinedInputClasses,
  tabClasses,
} from '@mui/material';

/* UKRI colours */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const UKRI_COLOURS = {
  blue: '#2E2D62',
  bright: {
    orange: '#FF6900', // pure orange
    yellow: '#FBBB10', // yellow
    green: '#67C04D', // light green
    blue: '#1E5DF8', // blue
    purple: '#BE2BBB', // bright purple
    red: '#E94D36', // light red
  },
  deep: {
    orange: '#C13D33', // pure orange
    yellow: '#F08900', // vivid yellow
    green: '#3E863E', // green
    blue: '#003088', // blue
    purple: '#8A1A9B', // bright purple
    red: '#A91B2E', // red
  },
};

/* Colours that may be used across light/dark modes e.g. the main app bar */
const STATIC_COLOURS = {
  darkBlue: '#003088',
  orange: '#C34F00',
};

/* Main colours used for dark/light modes respectively */
interface ThemeColours {
  /* The type of mode this is */
  type: 'default' | 'contrast';

  /* Primary/secondary colours used for MUI */
  primary: string;
  secondary: string;

  /* Secondary text colour */
  textSecondary: string;

  /* Background colours for the page and papers */
  background: string;
  paper: string;

  /* Standard colours used in plugins (change to lighter/darker shades
    between light and dark modes) - these are meant to give good contrast
    for text on the chosen paper background colour */
  blue: string;
  orange: string;
  red: string;
  grey: string;

  /* Lighter colours */
  lightBlue: string; //Used for lighter coloured buttons
  lightOrange: string; //Used for notifcation icon

  /* Darker colours */
  darkGreen: string; //Used for cookie consent message
  darkOrange: string; //Used for help tour

  /* Contrast colours that need to change significantly between dark
     and light modes */
  contrastGrey: string; //Used on homepage

  /* Informational/Warning colours */
  information: string; //Used in open data label
  warning: string; //Used for selection alert banner

  /* Colours for <a> style links */
  link: {
    default: string;
    visited: string;
    active: string;
  };

  footerLink: {
    default: string;
    active: string;
  };

  chip: string; //Used for chip colours in cards and filters

  /* Colours used on the homepage */
  homePage: {
    heading: string;
    blueDescription: string;
    blueButton: string;
  };

  /* Colour for Tabs in search */
  tabsGrey: string;
}

declare module '@mui/material/styles' {
  interface Theme {
    colours: ThemeColours;
    drawerWidth: string;
    footerPaddingTop: string;
    footerPaddingBottom: string;
    footerHeight: string;
    mainAppBarHeight: string;
  }
  // allow configuration using `createTheme`
  interface ThemeOptions {
    colours: ThemeColours;
    drawerWidth: string;
    footerPaddingTop: string;
    footerPaddingBottom: string;
    footerHeight: string;
    mainAppBarHeight: string;
  }
}

const DARK_MODE_COLOURS: ThemeColours = {
  type: 'default',
  primary: UKRI_COLOURS.deep.blue,
  secondary: '#80ACFF',
  textSecondary: 'rgba(255, 255, 255, 0.7)',
  background: '#1B1B1B',
  paper: '#3A3A3A',
  blue: '#86B4FF',
  orange: '#F26300',
  red: '#FF7F73',
  grey: '#A4A4A4',
  lightBlue: UKRI_COLOURS.bright.blue,
  lightOrange: '#FF6900',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#E6E6E6',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFA500',
  link: {
    default: '#86B4FF',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#FFFFFF',
    active: '#E58885',
  },
  chip: '#595959',
  homePage: {
    heading: '#FFFFFF',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#3A3A3A',
};

const DARK_MODE_HIGH_CONTRAST_COLOURS: ThemeColours = {
  type: 'contrast',
  primary: '#86B4FF',
  secondary: '#80ACFF',
  textSecondary: 'rgba(255, 255, 255, 0.7)',
  background: '#000000',
  paper: '#1A1A1A',
  blue: '#B4CCFA',
  orange: '#FFC14D',
  red: '#FFA198',
  grey: '#A4A4A4',
  lightBlue: UKRI_COLOURS.deep.blue,
  lightOrange: '#FFC14D',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#FFFFFF',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFC14D',
  link: {
    default: '#B4CCFA',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#000000',
    active: '#851D0F',
  },
  chip: '#3A3A3A',
  homePage: {
    heading: '#FFFFFF',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#1A1A1A',
};

const LIGHT_MODE_COLOURS: ThemeColours = {
  type: 'default',
  primary: UKRI_COLOURS.deep.blue,
  secondary: UKRI_COLOURS.deep.blue,
  textSecondary: 'rgba(0, 0, 0, 0.54)',
  background: '#FAFAFA',
  paper: '#FFFFFF',
  blue: UKRI_COLOURS.deep.blue,
  orange: '#C34F00',
  red: '#AC1600',
  grey: '#727272',
  lightBlue: UKRI_COLOURS.bright.blue,
  lightOrange: '#FF6900',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#717171',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFA500',
  link: {
    default: '#1E5DF8',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#FFFFFF',
    active: '#E58885',
  },
  chip: '#E0E0E0',
  homePage: {
    heading: '#333333',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#EEEEEE',
};

const LIGHT_MODE_HIGH_CONTRAST_COLOURS: ThemeColours = {
  type: 'contrast',
  primary: UKRI_COLOURS.deep.blue,
  secondary: UKRI_COLOURS.deep.blue,
  textSecondary: '#000000',
  background: '#FAFAFA',
  paper: '#FFFFFF',
  blue: '#002466',
  orange: '#C34F00',
  red: '#801100',
  grey: '#727272',
  lightBlue: UKRI_COLOURS.bright.blue,
  lightOrange: '#FF6900',
  darkGreen: '#3A7F3A',
  darkOrange: STATIC_COLOURS.orange,
  contrastGrey: '#000000',
  information: UKRI_COLOURS.deep.blue,
  warning: '#FFA500',
  link: {
    default: '#052d94',
    visited: '#BE2BBB',
    active: '#E94D36',
  },
  footerLink: {
    default: '#FFFFFF',
    active: '#E58885',
  },
  chip: '#E0E0E0',
  homePage: {
    heading: '#000000',
    blueDescription: '#FFFFFF',
    blueButton: UKRI_COLOURS.bright.blue,
  },
  tabsGrey: '#EEEEEE',
};

export const buildTheme = (
  darkModePreference: boolean,
  highContrastModePreference?: boolean,
  primaryColour?: string
): Theme => {
  const colours = darkModePreference
    ? highContrastModePreference
      ? DARK_MODE_HIGH_CONTRAST_COLOURS
      : DARK_MODE_COLOURS
    : highContrastModePreference
      ? LIGHT_MODE_HIGH_CONTRAST_COLOURS
      : LIGHT_MODE_COLOURS;

  if (primaryColour && !(darkModePreference && highContrastModePreference)) {
    colours.primary = primaryColour;
  }

  const componentOverrides = {
    MuiPaper: {
      styleOverrides: { root: { backgroundImage: 'unset' } },
    },
    MuiLink: {
      defaultProps: {
        underline: 'hover' as 'always' | 'none' | 'hover' | undefined,
      },
      styleOverrides: {
        root: {
          color: colours.blue,
        },
      },
    },
    MuiTabs: {
      styleOverrides: {
        indicator: {
          color: STATIC_COLOURS.darkBlue,
          textDecoration: 'underline',
        },
      },
    },
    MuiTab: {
      styleOverrides: {
        textColorSecondary: {
          [`&.${tabClasses.selected}`]: {
            color: darkModePreference ? '#FFFFFF' : colours.blue,
          },
        },
      },
    },
    MuiCheckbox: {
      styleOverrides: {
        colorPrimary: {
          [`&.${checkboxClasses.checked}`]: {
            color: colours.blue,
          },
          [`&.${checkboxClasses.indeterminate}`]: {
            color: colours.grey,
          },
        },
      },
    },
    MuiFormLabel: {
      styleOverrides: {
        root: {
          [`&.${formLabelClasses.error}`]: {
            color: colours.red,
          },
          [`&.${formLabelClasses.focused}`]: {
            color: colours.blue,
          },
        },
        asterisk: {
          [`&.${formLabelClasses.error}`]: {
            color: colours.red,
          },
        },
      },
    },
    MuiBadge: {
      styleOverrides: {
        colorPrimary: {
          backgroundColor: STATIC_COLOURS.orange,
          color: 'white',
        },
      },
    },
    MuiInput: {
      styleOverrides: {
        underline: {
          [`&.${inputClasses.error}`]: {
            borderBottomColor: colours.red,
          },
          [`&.${inputClasses.error}:after`]: {
            borderBottomColor: colours.blue,
          },
        },
      },
    },
    MuiOutlinedInput: {
      styleOverrides: {
        root: {
          [`&.${outlinedInputClasses.error} .${outlinedInputClasses.notchedOutline}`]:
            {
              borderColor: colours.red,
            },
        },
        //Override opacity of placeholder text
        input: {
          '&::placeholder': {
            colour: colours.textSecondary,
            opacity:
              !darkModePreference && highContrastModePreference ? 1.0 : 0.7,
          },
        },
      },
    },
    MuiFormHelperText: {
      styleOverrides: {
        root: {
          [`&.${formHelperTextClasses.error}`]: {
            color: colours.red,
          },
        },
      },
    },
    MuiChip: {
      styleOverrides: {
        root: {
          backgroundColor: colours.chip,
        },
      },
    },
    MuiPickersToolbar: {
      styleOverrides: {
        toolbar: {
          backgroundColor: colours.primary,
        },
      },
    },
    MuiPickersCalendarHeader: {
      styleOverrides: {
        dayLabel: {
          color: colours.grey,
        },
      },
    },
    MuiPickersDay: {
      styleOverrides: {
        current: {
          color: colours.blue,
        },
        dayDisabled: {
          color: colours.grey,
        },
      },
    },
    MuiPickersYear: {
      styleOverrides: {
        root: {
          '&:active': {
            color: colours.blue,
          },
          '&:focus': {
            color: colours.blue,
          },
        },
        yearSelected: {
          color: colours.blue,
        },
        yearDisabled: {
          color: colours.grey,
        },
      },
    },
    MuiPickersMonth: {
      styleOverrides: {
        root: {
          '&:active': {
            color: colours.blue,
          },
          '&:focus': {
            color: colours.blue,
          },
        },
        monthSelected: {
          color: colours.blue,
        },
        monthDisabled: {
          color: colours.grey,
        },
      },
    },
  };

  return createTheme({
    palette: {
      // Light/dark mode
      mode: darkModePreference ? 'dark' : 'light',
      primary: {
        main: colours.primary,
      },
      secondary: {
        main: colours.secondary,
      },
      text: {
        secondary: colours.textSecondary,
      },
      background: {
        default: colours.background,
        paper: colours.paper,
      },
    },
    typography: {
      button: {
        textTransform: 'none',
      },
    },
    components: componentOverrides,
    colours: colours,
    drawerWidth: '220px',
    footerPaddingTop: '8px',
    footerPaddingBottom: '8px',
    footerHeight: '20px',
    mainAppBarHeight: '64px',
  });
};

function mapThemeProviderStateToProps(state: StateType): {
  prefersDarkMode: boolean;
} {
  return {
    prefersDarkMode: state.scigateway.darkMode,
  };
}

const SciGatewayThemeProvider = (props: {
  children: React.ReactNode;
}): React.ReactElement<{
  children: React.ReactNode;
}> => {
  const darkModePreference = useSelector(
    (state: StateType) => state.scigateway.darkMode
  );
  const highContrastModePreference = useSelector(
    (state: StateType) => state.scigateway.highContrastMode
  );
  const primaryColour = useSelector(
    (state: StateType) => state.scigateway.primaryColour
  );
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider
        theme={buildTheme(
          darkModePreference,
          highContrastModePreference,
          primaryColour
        )}
      >
        {props.children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

export const ConnectedThemeProvider = connect(mapThemeProviderStateToProps)(
  SciGatewayThemeProvider
);
