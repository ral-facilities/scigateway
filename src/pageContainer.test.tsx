import React from 'react';

import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { createLocation } from 'history';
import { MemoryRouter } from 'react-router-dom';

import PageContainer from './pageContainer.component';
import { StateType } from './state/state.types';
import { authState, initialState } from './state/reducers/scigateway.reducer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from '@mui/material';
import { buildTheme } from './theming';

jest.mock('@mui/material', () => ({
  __esmodule: true,
  ...jest.requireActual('@mui/material'),
  useMediaQuery: jest.fn(() => true),
}));

describe('PageContainer - Tests', () => {
  let state: StateType;

  beforeEach(() => {
    state = {
      scigateway: { ...initialState, authorisation: { ...authState } },
      router: { location: createLocation('/') },
    };
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Provider store={configureStore([thunk])(state)}>
        <ThemeProvider theme={buildTheme(false)}>
          <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
            <PageContainer />
          </MemoryRouter>
        </ThemeProvider>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
