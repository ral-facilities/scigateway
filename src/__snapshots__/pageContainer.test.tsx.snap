// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer - Tests renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1kl99y5"
  >
    <div>
      <div
        class="MuiBox-root css-n0zxlq"
      >
        <div
          style="box-sizing: border-box; padding: 10px 0px;"
        >
          <div
            style="position: relative; display: block; margin: auto; width: 230px; height: 230px; animation: rotate 10s infinite linear;"
          >
            <i
              style="position: absolute; display: inline-block; top: 50%; left: 50%; border: 8px solid transparent; border-top-left-radius: 140px; border-top-right-radius: 140px; border-color: #8c4799; height: 70px; width: 140px; margin-top: -70px; margin-left: -70px; animation-name: rotate; animation-iteration-count: infinite; animation-duration: 3s; animation-timing-function: cubic-bezier(.09, 0, 0, .03); transform-origin: 50% 100% 0; box-sizing: border-box;"
            />
            <i
              style="position: absolute; display: inline-block; top: 50%; left: 50%; border: 8px solid transparent; border-top-left-radius: 148px; border-top-right-radius: 148px; border-color: #1d4f91; height: 79px; width: 158px; margin-top: -79px; margin-left: -79px; animation-name: rotate; animation-iteration-count: infinite; animation-duration: 3s; animation-timing-function: cubic-bezier(.09, 0.3, 0.12, .03); transform-origin: 50% 100% 0; box-sizing: border-box;"
            />
            <i
              style="position: absolute; display: inline-block; top: 50%; left: 50%; border: 8px solid transparent; border-top-left-radius: 156px; border-top-right-radius: 156px; border-color: #c34613; height: 88px; width: 176px; margin-top: -88px; margin-left: -88px; animation-name: rotate; animation-iteration-count: infinite; animation-duration: 3s; animation-timing-function: cubic-bezier(.09, 0.6, 0.24, .03); transform-origin: 50% 100% 0; box-sizing: border-box;"
            />
            <i
              style="position: absolute; display: inline-block; top: 50%; left: 50%; border: 8px solid transparent; border-top-left-radius: 164px; border-top-right-radius: 164px; border-color: #008275; height: 97px; width: 194px; margin-top: -97px; margin-left: -97px; animation-name: rotate; animation-iteration-count: infinite; animation-duration: 3s; animation-timing-function: cubic-bezier(.09, 0.8999999999999999, 0.36, .03); transform-origin: 50% 100% 0; box-sizing: border-box;"
            />
            <i
              style="position: absolute; display: inline-block; top: 50%; left: 50%; border: 8px solid transparent; border-top-left-radius: 172px; border-top-right-radius: 172px; border-color: #63666a; height: 106px; width: 212px; margin-top: -106px; margin-left: -106px; animation-name: rotate; animation-iteration-count: infinite; animation-duration: 3s; animation-timing-function: cubic-bezier(.09, 1.2, 0.48, .03); transform-origin: 50% 100% 0; box-sizing: border-box;"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-1jolfkj"
        >
          Loading...
        </div>
      </div>
    </div>
    <div
      style="width: 100%;"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-1p9tt43-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-iv6ml1-MuiToolbar-root"
        >
          <button
            aria-label="open-navigation-menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1nin6cb-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="home-page"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium tour-title css-bj8lf8-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <img
              alt="title"
              src="scigateway-white-text-blue-mark-logo.svg"
              style="height: 24px;"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            style="flex-grow: 1;"
          />
          <div
            class="tour-user-profile"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1yxmrfv-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1vvwrqo-MuiTypography-root"
              >
                login-button
              </p>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <button
            aria-haspopup="true"
            aria-label="open-mobile-menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-19c6lcd-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MoreVertIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
    <div
      class="react-joyride"
    />
    <div
      class="css-9peh1a"
    >
      <div
        id="dg-homepage"
      >
        <div
          style="background-image: url(background.jpg); width: 100%; height: 250px;"
        >
          <div
            style="background-repeat: no-repeat; width: 100%; height: 250px;"
          >
            <div
              class="MuiBox-root css-h95k7f"
            >
              <h2
                class="MuiTypography-root MuiTypography-h2 css-1qicgzl-MuiTypography-root"
              >
                <strong>
                  Data discovery
                </strong>
                 and 
                <strong>
                  access
                </strong>
              </h2>
              <h2
                class="MuiTypography-root MuiTypography-h2 css-1qicgzl-MuiTypography-root"
              >
                for 
                <strong>
                  large-scale
                </strong>
                science facilities
              </h2>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1qrn4l7"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-12uikys-MuiPaper-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              style="height: 100%;"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1onku40"
                >
                  <h3
                    class="MuiTypography-root MuiTypography-h3 css-vduuh3-MuiTypography-root"
                  >
                    home-page.browse.title
                  </h3>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-4y3rru-MuiTypography-root"
                  >
                    home-page.browse.description1
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-4y3rru-MuiTypography-root"
                  >
                    <strong>
                      DataGateway
                    </strong>
                     focuses on providing data discovery and data access functionality to the data.
                  </p>
                  <div
                    class="MuiBox-root css-1yuhvjn"
                  >
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-2ttl85-MuiButtonBase-root-MuiButton-root"
                      data-testid="browse-button"
                      href="/home-page.browse.link"
                      tabindex="0"
                    >
                      home-page.browse.button
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-12uikys-MuiPaper-root"
              >
                <div
                  class="MuiBox-root css-1onku40"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jftqyy-MuiAvatar-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-vyilrt-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                      />
                    </svg>
                  </div>
                  <h4
                    class="MuiTypography-root MuiTypography-h4 css-f630zx-MuiTypography-root"
                  >
                    home-page.search.title
                  </h4>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-4y3rru-MuiTypography-root"
                  >
                    home-page.search.description
                  </p>
                  <div
                    class="MuiBox-root css-8xl60i"
                  >
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-2ttl85-MuiButtonBase-root-MuiButton-root"
                      data-testid="search-button"
                      href="/home-page.search.link"
                      tabindex="0"
                    >
                      home-page.search.button
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-12uikys-MuiPaper-root"
              >
                <div
                  class="MuiBox-root css-1onku40"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jftqyy-MuiAvatar-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-vyilrt-MuiSvgIcon-root"
                      data-testid="GetAppIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 9h-4V3H9v6H5l7 7zM5 18v2h14v-2z"
                      />
                    </svg>
                  </div>
                  <h4
                    class="MuiTypography-root MuiTypography-h4 css-f630zx-MuiTypography-root"
                  >
                    home-page.download.title
                  </h4>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-4y3rru-MuiTypography-root"
                  >
                    home-page.download.description
                  </p>
                  <div
                    class="MuiBox-root css-8xl60i"
                  >
                    <a
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-2ttl85-MuiButtonBase-root-MuiButton-root"
                      data-testid="download-button"
                      href="/home-page.download.link"
                      tabindex="0"
                    >
                      home-page.download.button
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ju2rzk-MuiPaper-root"
              >
                <div
                  style="background-image: url(green-swirl2.png); background-repeat: no-repeat; background-position: top right; background-size: auto 100%; height: 100%;"
                >
                  <div
                    class="MuiBox-root css-1onku40"
                  >
                    <h4
                      class="MuiTypography-root MuiTypography-h4 css-94tke6-MuiTypography-root"
                    >
                      home-page.facility.title
                    </h4>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1f5ny8y-MuiTypography-root"
                    >
                      home-page.facility.description
                    </p>
                    <div
                      class="MuiBox-root css-8xl60i"
                    >
                      <a
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1h8ppsl-MuiButtonBase-root-MuiButton-root"
                        data-testid="facility-button"
                        href="home-page.facility.link"
                        tabindex="0"
                      >
                        home-page.facility.button
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
